concourse/pipelines/linux-image-build.jsonnet

```diff
@@ -385,6 +385,155 @@ local DebianImgPublishJob(image, env, workflow_dir) = imgpublishjob {
   image_prefix: common.debian_image_prefixes[image],
 };
 
+
+local saptestjob = {
+  local tl = self,
+
+  image:: error 'must be set',
+
+  name: 'sap-workload-test-' + self.image,
+
+  plan: [
+  {
+    get: tl.image + '-gcs',
+    passed: [
+      'publish-to-testing-' + tl.image,
+    ],
+    params: {
+      skip_download: 'true',
+    }
+  },
+  { get: 'guest-test-infra' },
+  {
+    task: 'generate-timestamp',
+    file: 'guest-test-infra/concourse/tasks/generate-timestamp.yaml',
+  },
+  {
+    load_var: 'id',
+    file: 'timestamp/timestamp-ms',
+  },
+  {
+    task: 'generate-post-script',
+    config: {
+      platform: 'linux',
+      image_resource: {
+        type: 'registry-image',
+        source: {
+          repository: 'google/cloud-sdk',
+          tag: 'latest',
+        },
+      },
+      inputs: [
+        { name: "guest-test-infra" },
+      ],
+      run: {
+        path: 'sh',
+        args: [
+          '-exc',
+          |||
+          cd guest-test-infra/concourse/scripts
+          # We want to upload this actual script with the unique id
+          sed -i 's/__BUCKET__/test-bucket-for-terraform/g' sap_post_script.sh
+          sed -i 's/__RUN__/((.:id))/g' sap_post_script.sh
+          gsutil cp sap_post_script.sh gs://test-bucket-for-terraform/workload-tests/sap/((.:id))/sap_post_script.sh
+|||,
+        ],
+      },
+    },
+  },
+  {
+    task: 'create-sap-tf-environment',
+    config: {
+      platform: 'linux',
+      image_resource: {
+        type: 'registry-image',
+        source: {
+          repository: 'hashicorp/terraform',
+          tag: 'latest',
+        },
+      },
+      inputs: [
+        { name: "guest-test-infra" },
+      ],
+      outputs: [
+        { name: "tf-state" },
+      ],
+      run: {
+        path: 'sh',
+        args: [
+          '-exc',
+          |||
+          cp guest-test-infra/concourse/scripts/sap_hana.tf tf-state/
+          cd tf-state
```

Comment:
this won't work, cp will complain

why make a copy? why not use `dir:` arg and avoid use of shell altogether?

Reply:
It looks like having these separate named outputs and inputs causes issues. If I set output as below and set `
        dir: 'guest-test-infra/concourse/scripts'`, the scripts folder will now be empty (the task won't see any files there).
```
      outputs: [
        { name: "tf-state", path: "guest-test-infra/concourse/scripts" },
      ],
```

I think setting the outputs field will init the folder and clear it out. Do you know a way to avoid this without doing this cp/cd stuff?